lib/ansible/playbook/role_include.py

```diff
@@ -131,7 +131,10 @@ def load(data, block=None, role=None, task_include=None, variable_manager=None,
         # validate bad args, otherwise we silently ignore
         bad_opts = my_arg_names.difference(IncludeRole.VALID_ARGS)
         if bad_opts:
-            raise AnsibleParserError('Invalid options for %s: %s' % (ir.action, ','.join(list(bad_opts))), obj=data)
+            raise AnsibleParserError('Invalid options for %s: %s valid options are: (%s)' % (ir.action,
+                                                                                             ','.join(list(bad_opts)),
+                                                                                             ','.join(list(IncludeRole.VALID_ARGS))),
```

Comment:
i would not display 'valid_args' here as they are not 'valid' in all cases of role invocation, this code gets used for both import/include_role and `roles:`

Reply:
Hmm, indeed. Removing for now. Maybe knowing/showing valid options for both cases is something to address in a different pr...